spec/SPEC-annex-C-source-embedding.md

```diff
@@ -0,0 +1,95 @@
+
+### Embedded Input Manifest Identifier
+
+Most source code files are hand coded by humans.  Some however are generated from other input(s) by a build tool.
+
+A build tool outputing a source code file may embed the Input Manifest Identifier for the output source code file into
+the output source code file by adding a comment line containing a string of the form:
+
+``` OmniBOR-Input-Manifest-ID: [ ${comma separated list of Input Manifest Identifier URIs} ]
+```
+
+For a file with C commenting semantics (like C, C++, Java, Go, etc) a concrete example might be:
+
+```// OmniBOR-Input-Manifest-ID: [ gitoid:blob:sha1:261eeb9e9f8b2b4b0d119366dda99c6fd7d35c64, gitoid:blob:sha256:09c825ac02df9150e4f93d12ba1da5d1ff5846c3e62503c814aa3a300c535772 ]
+```
+
+For a file with shell commenting semantics (like a shell script, Python, etc) a concrete example might be:
+```# OmniBOR-Input-Manifest-ID: [ gitoid:blob:sha1:261eeb9e9f8b2b4b0d119366dda99c6fd7d35c64, gitoid:blob:sha256:09c825ac02df9150e4f93d12ba1da5d1ff5846c3e62503c814aa3a300c535772 ]
+```
+
+### Placement of OmniBOR-Input-Manifest-ID Comment Line
+
+The OmniBOR-Input-Manifest-ID comment line should be placed as the last line in the source code file.  The OmniBOR-Input-Manifest-ID comment line should be proceeded by a blank line to insure it is not interpretted as part of another comment block.
+
+A tool reading the source code file should interpret the last OmniBOR-Input-Manifest-ID comment line it encounters in the file as being the Input Manifest Identifier, and ignore previous comment lines in the file which may contain Input Manifest Identifiers.
+
+Example:
+
+If the input source code file begins with:
+
+```go
+// Code generated by stringer DO NOT EDIT.
+
+import (
+    "fmt"
+)
+...
+```
+
+the output source code file should look like:
+```go
+// Code generated by stringer DO NOT EDIT.
+
+import (
+    "fmt"
+)
+...
+
+// OmniBOR-Input-Manifest-ID: [ gitoid:blob:sha1:261eeb9e9f8b2b4b0d119366dda99c6fd7d35c64, gitoid:blob:sha256:09c825ac02df9150e4f93d12ba1da5d1ff5846c3e62503c814aa3a300c535772 ]
+```
+
+if the input source code file begins with:
+
+```c
+/*
+ * Copyright 2023 Yoyodyne Inc
+ * SPDX-License-Identifier: 
+ */
+
+#include <stdio.h>
+int main() {
+   // printf() displays the string inside quotation
+   printf("Hello, World!");
+   return 0;
+}
+```
+
+the output source code file should look like:
+
+```c
+/*
+ * Copyright 2023 Yoyodyne Inc
+ * SPDX-License-Identifier: 
+ */
+
+#include <stdio.h>
+int main() {
+   // printf() displays the string inside quotation
+   printf("Hello, World!");
+   return 0;
+}
+
+//* OmniBOR-Input-Manifest-ID: [ gitoid:blob:sha1:261eeb9e9f8b2b4b0d119366dda99c6fd7d35c64, gitoid:blob:sha256:09c825ac02df9150e4f93d12ba1da5d1ff5846c3e62503c814aa3a300c535772 ] */
+```
+
+### Tools which mutate existing source code files
+
+Many source code generation tools, like patch, specifically mutate an existing input source code file which may contain
```

Comment:
Related: we should probably put out some guidance for how to deal with tools which don't fixup OmniBOR manifest identifiers themselves when they make edits, perhaps by guiding people to check for fixups, and providing some standard tool(s) to perform fixups on a per-language basis.